### ISSUE TITLE
Dark theme is Gtk3 Adwaita stylesheet in some Flatpaks

### ISSUE THREAD

**Issue by winkelnp** [lassekongo83/adw-gtk3#101]
In some flatpaks (like org.gnome.Evince or org.gnome.Lollypop) the dark theme shows up as the old adwaita style.

I do not have org.gtk.Gtk3Theme.Adwaita-dark installed, it is only your two flatpak themes and my Gtk3 theme is adw-gtk3(-dark).

Is this intended behaviour/ repeatable on your end? if not, what can I do?

Thanks for your help.

Top panel of org.gnome.Lollypop :
dark
![image](https://user-images.githubusercontent.com/68015877/166060942-2f6f9cf1-a955-4a51-ae01-d17b47167702.png)
light
![image](https://user-images.githubusercontent.com/68015877/166061038-dbabea30-0c1b-4f83-b9d0-6508f6dd0dc4.png)

**Comment by lassekongo83** [comment]
I can reproduce, but I have no idea why it happens. @dusansimic Any idea?

You can get the apps to use the dark theme though. If you installed the the theme in `~/.local/share/themes`:

```bash
sudo flatpak override --filesystem=$HOME/.local/share/themes
sudo flatpak override --env=GTK_THEME=adw-gtk3-dark
```

**Comment by dusansimic** [comment]
I was able to reproduce this on Lollypop (neither light or dark theme are working). I however don't think at the moment that this is the packages fault because it works as expected on many other gtk3 flatpak apps (GTG, Pitivi, all Electron apps...).

> Could it be because of the line rm adw-gtk3/gtk-3.0/gtk-dark.css

That line removes the dark theme css from light theme package because if it is left there and light theme is selected in gnome tweaks, it uses the dark theme for some reason. The dark theme is in the adw-gtk3-dark package and that one is loaded when dark theme is used.

I'll take a look at possible problems when I find time.

**Comment by winkelnp** [comment]
flatpak override resolves the issue, but then the apps don't follow system dark/light theme... (i know that's intended function)

Is this maybe an issue with how flatpak recognises GTK themes as opposed to the system dark theme? Maybe the system dark theme setting overrides any GTK themes that might have been set

**Comment by Microwave-WYB** [comment]
I had a the similar issue with flatpak apps. This can possibly solved by installing the theme using flatpak, see #116.

Try `flatpak install org.gtk.Gtk3theme.adw-gtk3  org.gtk.Gtk3theme.adw-gtk3-dark`

If that solves the problem, maybe we can update the README to make sure everyone follows this step.

**Comment by daudix-UFO** [comment]
Same thing, gnome web & bottles have same issue (flatpak)

**Comment by SkelyTony** [comment]
> I can reproduce, but I have no idea why it happens.
> 
> @dusansimic Any idea? Could it be because of the line `rm adw-gtk3/gtk-3.0/gtk-dark.css` in https://github.com/flathub/org.gtk.Gtk3theme.adw-gtk3/blob/master/org.gtk.Gtk3theme.adw-gtk3.yaml ?
> 
> You can get the apps to use the dark theme though. If you installed the the theme in `~/.local/share/themes`:
> 
> ```shell
> sudo flatpak override --filesystem=$HOME/.local/share/themes
> sudo flatpak override --env=GTK_THEME=adw-gtk3-dark
> ```

didn't fix the issue for me

**Comment by lassekongo83** [comment]
@[dusansimic](https://github.com/dusansimic) I've narrowed it down to that it in fact was the missing `gtk-dark.css` file. It is that file some applications use when switching between light/dark in `gnome-control-center`. If it is missing the app will default to classic Adwaita. It should be present in both the light and dark theme (even if it's a duplicate of gtk.css in the dark theme.)

Test it by downloading the tarball: 
1. Extract the themes to different named folders in `~/.local/share/themes`. For example `test-light`, `test-dark`.
2. Select `test-dark` in `gnome-tweaks`.
3. Remove `gtk-dark.css` from `test-light`.
4. Select dark in `gnome-control-center`.
5. Open a libhandy application like nautilus or gnome-calculator. It should now use classic Adwaita.

**Comment by SkelyTony** [comment]
> @dusansimic I've narrowed it down to that it in fact was the missing `gtk-dark.css` file. It is that file some applications use when switching between light/dark in `gnome-control-center`. If it is missing the app will default to classic Adwaita. It should be present in both the light and dark theme (even if it's a duplicate of gtk.css in the dark theme.)
> 
> Test it by downloading the tarball:
> 
>     1. Extract the themes to different named folders in `~/.local/share/themes`. For example `test-light`, `test-dark`.
> 
>     2. Select `test-dark` in `gnome-tweaks`.
> 
>     3. Remove `gtk-dark.css` from `test-light`.
> 
>     4. Select dark in `gnome-control-center`.
> 
>     5. Open a GTK3 libhandy application like nautilus. It should now use classic Adwaita.

so how do we fix this

**Comment by lassekongo83** [comment]
> so how do we fix this

Simply wait for it to be fixed in the flatpak packages. I'm not the one maintaining it.

**Comment by dusansimic** [comment]
@lassekongo83 thanks for that info! I had some troubles when adding both `gtk.css` and `gtk-dark.css` to the dark theme iirc so that's why I removed it from the dark theme (just renamed the `gtk-dark.css` to `gtk.css`). I'll look into it tonight or tomorrow. And see if I can fix it.

**Comment by dusansimic** [comment]
On my machine, with the 2.0 update, everything seems to work fine. Once the update gets propagated to flathub, it should probably work. Let's keep this issue open just in case.

**Comment by FiestaLake** [comment]
This still happens in v2.0 with flatseal (flatpak) app.

**Comment by daudix-UFO** [comment]
@FiestaLake flatpak theme is not updated for now, wait when it will update, because issue already fixed (I hope)

**Comment by FiestaLake** [comment]
> @FiestaLake flatpak theme is not updated for now, wait when it will update, because issue already fixed (I hope)

I think it already got updated a day ago.

**Comment by winkelnp** [comment]
Update to 2.0 was pushed to the flathub version on the 31st of May

**Comment by daudix-UFO** [comment]
Yes, I checked out, flatpak theme is updated but bug is still there, I have it in epiphany and flatseal, other gtk3 apps like dialect, curtail, fractal, etc is look normal

**Comment by STBoyden** [comment]
Can confirm, also experiencing this issue with multiple Flatpaks.

**Comment by winkelnp** [comment]
I get the feeling it affects more flatpaks now than before...

**Comment by BigmenPixel0** [comment]
Can confirm this issue.

**Comment by daudix-UFO** [comment]
I noticed strange thing, when installing flatpak theme with [stylepak](https://github.com/refi64/stylepak) there is no issues with not themed apps (flatseal, GNOME web, dialect, celluloid & etc)

**Comment by dusansimic** [comment]
I'm not sure why but for some reason everything works fine on my machine. I'm using Arch and have installed bot Flatpak theme and the system one (via the aur package). I've selected the system theme in Gnome Tweaks and the Flatpak theme is selected automatically after that.

**Comment by user1-github** [comment]
Same issue here. On Fedora Silverblue, the preinstalled utilities that still use gtk3, use the old dark Adwaita theme instead of adw-gtk3-dark.

**Comment by winkelnp** [comment]
the latest update to the flatpak seems to have fixed it for me. Does anyone else see it fixed now? If yes I'd be happy to mark this as resolved

**Comment by lassekongo83** [comment]
Run `flatpak update org.gtk.Gtk3theme.adw-gtk3` to resolve the issue.

**Comment by lassekongo83** [comment]
@user1-github Do you have any GTK_THEME environment variable set? You may have to unset it. `sudo flatpak override --unset-env=GTK_THEME`

**Comment by user1-github** [comment]
@lassekongo83 I don't have any env variables.

I've set up my theme in this order:
-Set the dark appearance in gnome settings
-Extract both light and dark non-Flatpak variants of the theme from the tarball to /.local/share/themes
-Set the non-Flatpak variant with this command: gsettings set org.gnome.desktop.interface icon-theme 'adw-gtk3-dark'
-Install both light and dark variants of the Flatpak theme.

Rebooting also didn't fix the issue after the update.

**Comment by lassekongo83** [comment]
@user1-github I just noticed you mentioned Document viewer. That is a GTK4 app, so you'll need to extract the libadwaita theme for that. [I have instructions on that](https://github.com/lassekongo83/adw-gtk3/blob/main/gtk4.md), but not for Silverblue.

I also got the Image viewer to use the theme when I put it in `~/.themes` instead of `~/.local/share/themes`. Must be something Silverblue specific.

**Comment by user1-github** [comment]
@lassekongo83 It works, thanks!
After I moved the themes to ~/.themes, both the Image Viewer and Document Viewer Flatpaks started using the adw-gtk3-dark theme.

**Comment by PhilDevProg** [comment]
I have exactly this issue but only with GNOME Web on Fedora. Every Flatpak App picks up adw-gtk3-dark except Web...
